azkaban-common/src/main/java/azkaban/imagemgmt/rampup/ImageRampupManagerImpl.java

```diff
@@ -166,8 +173,12 @@ public ImageRampupManagerImpl(final ImageRampupDao imageRampupDao,
   public Map<String, VersionInfo> getVersionByImageTypes(final ExecutableFlow flow,
       final Set<String> imageTypes, Set<String> overlayImageTypes)
       throws ImageMgmtException {
-    final Map<String, List<ImageRampup>> imageTypeRampups = this.imageRampupDao
-        .getRampupByImageTypes(imageTypes);
+    // If the flow is high priority flow, then skip fetching rampup plan and
+    // use the ACTIVE version of the image.
+    final boolean isHPFlow = this.hpFlowDao.isHPFlow(flow);
+    final Map<String, List<ImageRampup>> imageTypeRampups = isHPFlow ?
+        new LinkedHashMap<>(1) :
+        this.imageRampupDao.getRampupByImageTypes(imageTypes);
     final Set<String> remainingImageTypes = new TreeSet<>();
     final Map<String, ImageVersionMetadata> imageTypeVersionMap =
         this.processAndGetVersionForImageTypes(flow, imageTypes, imageTypeRampups,
```

Comment:
Just want to verify, in case of hp flow, processAndGetVersionForImageTypes will give active version?

Reply:
Yes.